### PR TITLE
Smarter image processing and skeleton loader

### PR DESCRIPTION
**Pull request by fairnightzz**

### CODE REVIEW

File: src/utils/requests.tsx [hunk 1]
```diff
@@ -4,30 +4,41 @@ import { API_URL } from 'config'
...
+    }
+    return result;
+  }, []);
+  return chapters;
+};
+
+export const getImage = async (url: string): Promise<string> => {
+
+  const { data } = await axios.get(url, { responseType: 'arraybuffer' });
+  console.log('yataa')
```
Comment (fairnightzz) [inline src/utils/requests.tsx]:
```suggestion
```

**Comment by MrPicklePinosaur** [comment]
did you have to branch off #11 for this MR

makes reviewing it a bit weird since i also see changes from the other MR. might be better to merge this branch into `local-storage` when creating MR

also u should delete the branches after merging

**Comment by MrPicklePinosaur** [comment]
by MR i mean PR

File: src/pages/Reader.tsx [hunk 2]
```diff
@@ -6,15 +6,17 @@ import { InView } from 'react-intersection-observer';
...
-import { getPages, getChapters } from 'utils/requests';
+import { getPages, getChapters, getImage } from 'utils/requests';
 import queryString from 'query-string';
 import { API_URL } from 'config';
 
 interface State {
   zipped: ZipInfo[],
-  images: (string | null)[],
+  images: (string)[],
+  imageFiles: (string)[],
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx]: why not `string[]`

File: src/utils/requests.tsx [hunk 3]
```diff
@@ -4,30 +4,41 @@ import { API_URL } from 'config'
...
-      }
-      return result;
-    }, []);
-    return chapters;
+  const url = `${API_URL}/${mangafile}/jpg`;
+  let { data: chapters } = await axios.get(url);
+
+  // filter out useless stuff
+  chapters = chapters.reduce((result: Chapter[], chapter: Record<string, unknown>) => {
+    const name = chapter.name as string;
```
Comment (MrPicklePinosaur) [inline src/utils/requests.tsx:26]: can chapter.name be null for any reason?

**Review by MrPicklePinosaur — APPROVED** [review]
lgtm more or less

File: src/utils/requests.tsx (hunk 3 above)
Comment (fairnightzz) [reply src/utils/requests.tsx:26]: no because that would mean you have no file name (i think)

**Pull request merged**